flocker/node/agents/test/test_blockdevice.py

```diff
@@ -2310,7 +2310,12 @@ def test_profile_respected(self):
         mandatory profiles.
         """
         for profile in (c.value for c in MandatoryProfiles.iterconstants()):
-            self.api.create_volume_with_profile(profile)
+            size = GiB(1).to_Byte().value
+            dataset_id = uuid4()
+            self.addCleanup(detach_destroy_volumes, self.api)
+            self.api.create_volume_with_profile(dataset_id=dataset_id,
+                                                size=size,
+                                                profile_name=profile)
 
```

Comment:
Thanks for your thoughts, @itamarst @sarum90 ! 

Made a few enhancements to interface based on your feedback and expected usage of profiles. Will schedule a meeting thursday morning to hear your thoughts on the new revision. Thanks!